src/components/HelloWorld.vue

```diff
@@ -1,57 +1,212 @@
+      Submitted Details:
+      <table>
+        <thead>
+          <tr>
+            <th>S.No</th>
+            <th @click="orderBy('name')">name</th>
+            <th @click="orderBy('email')">email</th>
+            <th @click="orderBy('dob')">DOB</th>
+            <th colspan="2">Action</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr v-for="user in users" :key="user.id" id="list">
+            <td> <span>{{user.id}}</span></td>
+            <td> <span>{{user.name}}</span></td>
+            <td> <span>{{user.email}}</span> </td>
+            <td> <span>{{user.dob}}</span> </td>
+            <td><a href="#" @click="deleteItem(user)">Delete</a>
+              <a href="#" @click="edit(user)" v-b-modal.modalPrevent>Edit</a></td>
+          </tr>
+        </tbody>
+      </table>
+  
+    </div>
+    <!-- Modal Component -->
+    <b-modal id="modalPrevent" ref="modal" title="Submit your Detail" @ok="handleOk" @shown="clearName">
+      <form @submit.stop.prevent="handleSubmit">
+        <b-form-input class="input" type="text" placeholder="Enter your Name" v-model="user.name"></b-form-input>
+        <b-form-input class="input" type="email" placeholder="Enter your Email" v-model="user.email"></b-form-input>
+        <b-form-input class="input" type="date" placeholder="Enter your DOB" v-model="user.dob"></b-form-input>
+        <b-form-input class="input" type="password" placeholder="Enter your Password" v-model="user.password"></b-form-input>
+        <b-form-input class="input" type="password" placeholder="Enter your Confirm-Password" v-model="user.con_password"></b-form-input>
+        <b-form-checkbox id="checkbox1" v-model="status" value="accepted" unchecked-value="not_accepted">
+          I accept the terms and use
+        </b-form-checkbox>
+      </form>
+    </b-modal>
   </div>
 </template>
 
 <script>
-export default {
-  name: 'HelloWorld',
-  props: {
-    msg: String
-  }
-}
+  export default {
+    name: "Form",
+    data() {
+      return {
+        users: [],
+        userCount: 0,
+        status: false,
+        user: {
+          email: "",
+          name: "",
+          dob: "",
+          password: "",
+          con_password: ""
+        },
+        isEdit: false,
+        sortKey: "",
+        sortSettings: [{
+            email: true
+          },
+          {
+            name: true
+          },
+          {
+            dob: true
+          }
+        ],
+        desc: true
+      };
+    },
+    methods: {
+      clearName() {
+        this.name = "";
+      },
+      clearEmail() {
+        this.email = "";
+      },
+      clearDob() {
+        this.dob = "";
+      },
+      clearPassword() {
+        this.password = "";
+      },
+      clearConfirmPassword() {
+        this.con_password = "";
+      },
+      handleOk(evt) {
+        // Prevent modal from closing
+        evt.preventDefault();
+        if (!this.user.email) {
+          alert("Please enter your email");
```

Comment:
don't use alert, since you are using bootstrap.

so inline error messages as supported by bootstrap